client/my-sites/site-settings/redirect-to-general.jsx

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import page from 'page';
+import { flowRight } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { isSiteAutomatedTransfer } from 'state/selectors';
+
+const redirectNonJetpackToGeneral = ( ComponentToRender ) => {
+	class RedirectNonJetpackToGeneral extends Component {
+		static propTypes = {
+			siteSlug: PropTypes.string,
+		}
+
+		componentDidMount() {
+			this.verifySiteIsJetpack();
+		}
+
+		componentDidUpdate() {
+			this.verifySiteIsJetpack();
+		}
+
+		verifySiteIsJetpack() {
+			if ( this.props.siteIsJetpack === false || this.props.siteIsAtomic ) {
+				this.redirectToGeneral();
+			}
+		}
+
+		redirectToGeneral = () => {
+			const {
+				siteSlug,
+			} = this.props;
+
+			if ( ! siteSlug ) {
+				return null;
+			}
+			page( '/settings/general/' + siteSlug );
```

Comment:
I don't see a reason to `return null` here. Would be shorter if we invert the condition:

```
if ( siteSlug ) {
	page( '/settings/general/' + siteSlug );
}
```

Reply:
:)